pool.go

```diff
@@ -298,31 +372,42 @@ func (p *pool) doFetch(ctx context.Context, from string, c cid.Cid) (b blocks.Bl
 		}
 	}
 
-	resp, err := p.config.Client.Do(req)
+	resp, err := p.config.SaturnClient.Do(req)
 	if err != nil {
 		return nil, err
 	}
+	defer resp.Body.Close()
+	if resp.StatusCode != http.StatusOK {
+		return nil, ErrBackendFailed
+	}
+
 	fb = time.Now()
 	code = resp.StatusCode
 	proto = resp.Proto
 	respReq = resp.Request
-	defer resp.Body.Close()
-	rb, err := io.ReadAll(resp.Body)
-	if err != nil {
-		return nil, err
+
+	buf := make([]byte, maxBlockSize)
+	received, err = io.ReadFull(resp.Body, buf)
+	// ReadFull returns an EOF ONLY if no bytes were read.
+	if err == io.EOF {
+		return nil, ErrBackendFailed
+	}
+	// If we read less than max bytes, we are good here.
+	if received > 0 && received <= maxBlockSize {
+		buf = buf[:received]
+	} else {
+		return nil, ErrBackendFailed
```

Comment:
This should be a separate error – otherwise we will bang our heads why things do not work N years from now, if ever the  block size policy changes (but fine to do it in https://github.com/filecoin-saturn/caboose/issues/28)

Reply:
Yeah will do it in #28